### PR TITLE
Better error logging on the client side

### PR DESCRIPTION
**Pull request by mikito-stripe**
r? @cjavilla-stripe

As mentioned earlier, I did a test run setting up https://github.com/stripe-samples/checkout-single-subscription and took notes (happy to share via slack).

One of the issues I noticed was that if I didn't fill in all the values in `.env`, the errors I got on the client side were a bit misleading because we were not logging all the errors.

This PR adds a bit more error handling to `client.js`, resulting in better error messages when an error occurs when fetching `/setup` or `/create-checkout-session`.

Error before:

Screen Shot 2020-10-07 at 10 54 26 AM

Error after:

Screen Shot 2020-10-07 at 10 54 15 AM

### CODE REVIEW

File: client/script.js
Comment (cjavilla-stripe) [inline client/script.js:6]: I think we'll likely need to figure out the best way to do this for all samples. I've kinda been thinking we may want to do these sorts of things inline? Since docs writers typically copy and paste the bits from these pages which are relevant to the docs and I worry they might not copy this context? 

Also wondering if we want to alert the error message or something. wdyt?

Comment (mikito-stripe) [reply client/script.js:6]: yeah I think you're right that it would be great to have a more general solution to this!

I would imagine that getting a misleading error is one of the ways in which folks might get blocked when getting started with the samples, so solving the problem in general would be good.

I don't think this code can be made much shorter - we kind of need the error from the JSON message and the `fetch` API requires that we get it in ~roughly this way - so I think the remain question is, how do we best incorporate  this snippet here and in other examples.

I think the options here are:

1) have a separate error handling function like the current PR

- Pro: sample code is easier to read
- Pro: error handling is centralized / avoids repeating this code twice
- Con: might risk docs writers not including the error handling

2) repeat this error handling function inline in every `fetch` invocation

- Pro: reduces the likelihood that docs writes miss the error handling
- Con: repeats the same code many times
- Con: code gets longer / more difficult to read

I would actually pick option 1 over option 2, here's why:

I don't think that "Con: might risk docs writers not including the error handling" is a big enough deal to sacrifice sample code readability. 

The worst case scenario isn't that bad -  even if the docs snippets don't have error handling, I think it'll be OK because following a docs tutorial is a different experience from running and then adapting a ready made sample:

As a developer, if I'm going through the docs example, I'm probably adapting the code in some way already and am more aware of what is going on, so if an error is thrown and not handled, I will probably spot it.

Like I'd imagine a docs walkthrough is more useful if you already have a server implementation and are just adding the client side stuff, in which case you already know a bit about the things that could go wrong on the server. Versus, if I'm running a sample I don't typically read the code before I start running the code, so better error messages are absolutely useful to help me figure out what code is worth reading.

So since I think it OK for docs not to do perfect error handling, I'd rather keep extract the code in a separate function and keep the sample code readable for sample code users, and leave it up to the docs writers as far as how much error handling to include in the examples on the docs site.

> Also wondering if we want to alert the error message or something. wdyt?

Not sure what you mean by "alert" here - the error message already shows up in the console when an error is thrown.

Comment (cjavilla-stripe) [reply client/script.js:6]: Okay yeah thats fine. by "alert" I meant the literal javascript alert function :)

That way if someone doesn't know how to open the console they see the error in the oldschool popup. I'm fine not blocking on this.

**Review by cjavilla-stripe — APPROVED** [review]
LGTM

**Pull request merged**
